src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterFactory.cs

```diff
@@ -40,9 +40,6 @@ internal sealed override ClassType ClassType
 
         internal override JsonPropertyInfo CreateJsonPropertyInfo()
         {
-            // We should never get here.
-            Debug.Assert(false);
```

Comment:
Why isn't this assert helpful?  If we hit this, it would cause a fail fast, whereas the exception could get eaten / handled higher up the stack.  Though it'd be more helpful with a message, e.g. `Debug.Fail("We should never get here.");`